ASTProcessor/src/main/java/edu/usb/argos/ASTProcessor/bestpractices/analyzer/BlockAnalyzer.java

```diff
@@ -0,0 +1,96 @@
+package edu.usb.argos.ASTProcessor.bestpractices.analyzer;
+
+import edu.usb.argos.ASTProcessor.antlr.JavaParser;
+import edu.usb.argos.ASTProcessor.bestpractices.HardcodedDetection;
+import edu.usb.argos.ASTProcessor.bestpractices.HardcodedValueMatcher;
+import lombok.AllArgsConstructor;
+
+import java.util.List;
+
+@AllArgsConstructor
+public class BlockAnalyzer {
+    private final HardcodedValueMatcher matcher;
+    private static BlockAnalyzer instance;
+
+    public static synchronized BlockAnalyzer getInstance() {
```

Comment:
What is the need to use Singleton pattern? It can cause bottlenecks with many analysis at the same time.